pkg/querier/metadata_handler.go

```diff
@@ -46,7 +48,32 @@ type metadataErrorResult struct {
 // Mimir for a given tenant. It is kept and returned as a set.
 func NewMetadataHandler(m MetadataSupplier) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		resp, err := m.MetricsMetadata(r.Context())
+		limit := -1
```

Comment:
How does prometheus work with the limit of 0? 

A lot of other limits in mimir we use 0 as the no-limit value, it would be nice if we can do it here too. Then we can safely use the zero-valued MetricMetadataOptions (or MetricMetadataRequest) to signal that there wasn't a limit provided.

Reply:
For consistency with Prometheus, currently Mimir mimics the exact same behaviour as in https://github.com/prometheus/prometheus/blob/c5e1498ae243bbf0b8363786de065bf98acd1edb/web/api/v1/api.go#L1221 which is actually internally inconsistent as 0 for limit per metric makes it ignored, while 0 for overall limit makes it return no metrics. If this is a problem, maybe we need to open a PR to fix this upstream first. For now, we have a default request method which sets the parameters to the ignored -1 values.

Reply:
I think consistency at least between the two values would be nice, it could be a bug in prometheus. 

So if we're mimicking the behaviour in prometheus, can you add some tests that catch these edge cases? like negative values and 0?